File: simple-stack-example-mvp/src/main/java/com/zhuinden/simplestackdemoexamplemvp/util/SchedulerHolder.java
package com.zhuinden.simplestackdemoexamplemvp.util;

import javax.inject.Inject;

import rx.Scheduler;

/**
 * Created by Owner on 2017. 01. 26..
 */
public class SchedulerHolder {
    private Scheduler scheduler;

    @Inject
    public SchedulerHolder() {
    }

    public Scheduler getScheduler() {
        if(scheduler == null) {
            throw new IllegalStateException("Scheduler should not be null!");
        }
        return scheduler;
    }

    public void setScheduler(Scheduler scheduler) {
        this.scheduler = scheduler;
    }
}

File: simple-stack-example-mvp-view/src/main/java/com/zhuinden/simplestackdemoexamplemvp/util/SchedulerHolder.kt
package com.zhuinden.simplestackdemoexamplemvp.util

import io.reactivex.Scheduler
import javax.inject.Inject


/**
 * Created by Owner on 2017. 01. 26..
 */
// NOT SCOPED!
class SchedulerHolder @Inject constructor() {
    lateinit var scheduler: Scheduler
}
